src/permitInfo/calculatePermitInfo.ts

```diff
@@ -0,0 +1,154 @@
+import {getTokenPermitInfo, PermitInfo} from 'alfetopito-permit-utils' // import {getTokenPermitInfo, PermitInfo} from '@cowprotocol/permit-utils' // TODO: maybe make the args nicer?
+import * as path from 'node:path'
+import {ethers} from 'ethers'
+import {readFileSync, writeFileSync} from 'node:fs'
+import {JsonRpcProvider} from '@ethersproject/providers'
+
+// TODO: maybe make the args nicer?
+// Get args from cli: chainId, optional token lists path, optional rpcUrl
+const [, , chainId, tokenListPath, rpcUrl] = process.argv
+
+const BASE_PATH = path.join('src/public/')
+
+async function calculatePermitInfo(
+  chainId: number,
+  tokenListPath: string | undefined,
+  rpcUrl: string | undefined,
+): Promise<void> {
+  // Load existing permitInfo.json file for given chainId
+  const permitInfoPath = path.join(BASE_PATH, `PermitInfo.${chainId}.json`)
+
+  let allPermitInfo: Record<string, PermitInfo> = {}
+
+  // Load existing permitInfo.json file for given chainId if it exists
+  try {
+    allPermitInfo = JSON.parse(readFileSync(permitInfoPath, 'utf8')) as Record<string, PermitInfo>
+  } catch (e) {
+    console.info(`File ${permitInfoPath} not found. It'll be created.`, e)
+  }
+
+  // Build provider instance
+  const provider = getProvider(chainId, rpcUrl)
+
+  // Load tokens info from a token list
+  const tokens = getTokens(chainId, tokenListPath)
+
+  // Create a list of promises to check all tokens
+  const promises = tokens.map((token) => {
+    const existingInfo = allPermitInfo[token.address.toLowerCase()]
+
+    return fetchPermitInfo(chainId, provider, token, existingInfo)
+  })
+
+  // Await for all of them to complete
+  const results = await Promise.allSettled(promises)
+
+  // Iterate over each result
+  results.forEach((result) => {
+    // Ignore failed or the ones where the value is falsy
+    if (result.status === 'fulfilled' && result.value) {
```

Comment:
I think we should use Promise.all becasue this fails silently and overrides the file with a subset of the values we want.

Or maybe is not a big deal as this will add data / update the old doc.

Reply:
Technically yes, the original file is overwritten, but all the original info is still there.
Whatever new data is fetched, is added to the permits map.
And it does not check for tokens which we have an info already.

I used `allSettled` because it's an expensive/slow operation, and in case of promise fails, all the successful results are thrown away.

Still, I don't mind changing if you feel it makes more sense.

Reply:
no, it fine